app/src/main/java/app/revanced/integrations/sponsorblock/ShieldButton.java

```diff
@@ -92,23 +99,32 @@ public static void changeVisibility(boolean visible, boolean immediate) {
                 LogHelper.printDebug(() -> "Fading out");
                 if (!immediate)
                     iView.startAnimation(fadeOut);
-                iView.setVisibility(shouldBeShown() ? View.INVISIBLE : View.GONE);
+                iView.setVisibility(View.GONE);
             }
         } catch (Exception ex) {
             LogHelper.printException(() -> "changeVisibility failure", ex);
         }
     }
 
-    static boolean shouldBeShown() {
+    private static boolean shouldBeShown() {
         return SettingsEnum.SB_ENABLED.getBoolean() && SettingsEnum.SB_NEW_SEGMENT_ENABLED.getBoolean();
     }
 
-    //region Helpers
-    private static int getIdentifier(String name, String defType) {
-        Context context = ReVancedUtils.getContext();
-        return context.getResources().getIdentifier(name, defType, context.getPackageName());
+    public static void hide() {
+        if (!isShowing) {
+            return;
+        }
+        ReVancedUtils.verifyOnMainThread();
+        View v = buttonReference.get();
+        if (v == null) {
+            return;
+        }
+        v.setVisibility(View.GONE);
+        isShowing = false;
     }
 
+    //region Helpers
+
     private static int getInteger(String name) {
```

Comment:
This belongs to `ReVancedUtils`. Theres other classes which require it as well.

Reply:
Yes, there are couple of other methods which have their own `getInteger` method.